prometheus.lua

```diff
@@ -348,7 +396,24 @@ function Prometheus:inc(name, label_names, label_values, value)
   -- Hopefully this does not happen too often (shared dictionary does not get
   -- reset during configuation reload).
   if err == "not found" then
-    self:set(key, value)
+    self:set_key(key, value)
+    return
+  end
+  -- Unexpected error
+  self:log_error_kv(key, value, err)
+end
+
+-- Set the current value of a gauge to `value`
+-- Args:
+--   name: (string) short metric name without any labels.
+--   label_names: (array) a list of label keys.
+--   label_values: (array) a list of label values.
+--   value: (number) the new value for the gauge.
+function Prometheus:set(name, label_names, label_values, value)
+  local key = full_metric_name(name, label_names, label_values)
+
+  local success, err = self.dict:set(key, value)
```

Comment:
`dict:set` might succeed by evicting some items from the dictionary, which is undesirable (since it means losing data). That's why `safe_set` is used instead in `set_key()`.

I think `Prometheus:set()` can look as simple as:
```
function Prometheus:set(name, label_names, label_values, value)
  local key = full_metric_name(name, label_names, label_values)
  self:set_key(key, value)
end
```